tests/ops/test_qubit_ops.py

```diff
@@ -394,6 +398,73 @@ def circuit(basis_state):
             circuit(basis_state)
 
 
+class TestSparse:
+    """Tests for sparse hamiltonian observable"""
+
```

Comment:
Should we place this in it's own file? I suddenly realize that it might be hard for developers in the future to find these tests.

Reply:
We are thinking of collecting all sparse-related functions in one place in the future. This will be useful if we want to add more functionality such as supporting the adjoint differentiation method. If this is ok for now, we can keep the tests here, otherwise I can simply create a test file and collect all sparse tests there.

Reply:
Done.